docs/proposals/20230612-node-os-image-caching.md

```diff
@@ -116,74 +118,78 @@ Example of the enviornment variable being turned on:
-
-In terms of how a healthy node would be selected, there is already a timestamp on each AzureMachinePool from its status.conditions.lastTransitionTime type ScaleSetModelUpdated as to when they’ve last been patched. There is also a timestamp on its status.conditions.lastTransitionTime type ScaleSetRunning as to when it last started running. For AzureMachine instances, we would need to add these timestamps or track it programmatically, below I have put an example of the yaml changes for storing the timestamp. An ideal node would be one which has been patched since the last prototype went into service and is running and healthy. Whichever node has been running and healthy for the longest amount of time since the last patch and has the patch applied should be chosen as it’s the most overall stable. As the prototype is always from a successfully healthy and working node the image is always known to be working before being chosen for replication.
+When the process is started it should go through the nodes of the cluster, choose a healthy node, shut it down, take a snapshot of it, restart it, create a compute image gallery image, delete the snapshot, and then configure the AzureMachineTemplate specs to use that compute image gallery image. After, it will store the current time as its timestamp. As a note, for the first implementation of this feature we will require the user to also use a compute image gallery image.
 
 Diagram of the Node OS Caching Process:
 
 ![Figure 1](./images/node-os-image-cache.png)
 
-Example AzureMachinePool yaml with the timestamps in status:
+As for why the healthy node has to be shut down while creating a snapshot of it, if it isn’t shut down first then pods can be scheduled as the snapshot is taken which will cause some dangerous states in terms of how it exists after being utilized by the AzureMachineTemplates.
+
+In terms of how a healthy node would be selected, there is already state data present on each AzureMachinePoolMachine and AzureMachine which we can use and is listed in the examples below. An ideal node would be one which is running and healthy. Whichever node has been running and healthy for the longest amount of time since the last patch and has the patch applied should be chosen as it’s the most overall stable. This means that for AzureMachinePoolMachines and AzureMachines we will take the node with the earliest creation time from metadata.creationTimestamp and has latestModelApplied : true present if it is an AzureMachinePoolMachine. As the prototype is always from a successfully healthy and working node the image is always known to be working before being chosen for replication. In terms of knowing if the patch has been applied, for the first development of this feature we won't know but all nodes should roughly be patched together so it will be fine in most cases.
```

Comment:
> Whichever node has been running and healthy for the longest amount of time since the last patch and has the patch applied should be chosen as it’s the most overall stable. This means that for AzureMachinePoolMachines and AzureMachines we will take the node with the earliest creation time from metadata.creationTimestamp and has latestModelApplied : true present if it is an AzureMachinePoolMachine.

I'm not sure this paragraph makes clear the difference between "patches" and `latestModelApplied` (or if there is any difference). To me it reads like we might be using one? or the other? or both?

And it's not also clear to me based on the doc exactly what "patches" are in this context.

Reply:
Tried to clarify latestModelApplied definition here and remove use of "patches" in an ambiguous sense and only use it where explicit (so automatic kernel patches).